### DISCUSSION TITLE
Request to FreeTube developers: please work with UI experts

### DISCUSSION THREAD

**Discussion by pkrasicki** [FreeTubeApp/FreeTube#4946] [General]
FreeTube is an amazing app, but its UI is very outdated in some places [compared to similar applications](https://github.com/FreeTubeApp/FreeTube/issues/3918#issuecomment-1707975145). I've seen developers implement extra buttons or settings as a workaround for those issues. This helps, but only partially and often at the cost of making the UI more complicated. There is only one proper solution: better design. Some parts of the UI have to be redesigned bit by bit. It can be done gradually, so that users have time to adjust to changes.

Developers, please work with UI experts to improve the UI. Take their opinions into account when developing new features. Please don't repeat the mistakes of many other Free Software projects. I know that there is a lot of other important work to do, but UI is also very important. Currently it feels like there is little focus in this area and that the design hasn't changed much in the last few years.

**Comment by kommunarr** [comment]
Hi @pkrasicki, I'm a UX-inclined developer who has been working on FT in the bouts I'm not on hiatus. I certainly have a lot to learn, but I'm trying alongside others to make improvements to the user experience wherever possible (#4929, #4921, WIP grid view for User Playlists). We're volunteers working to do what we can whenever we have the available time, and we welcome all the support we can get. If you or anyone with UI/UX knowledge has any evidence-backed suggestions and development efforts to help make the project better, we strongly welcome that. Just remember that this is not a private or for-profit project, where where we can "bring someone in" without their willing & unpaid support; people chip in where they can to make something better.

**Reply by pkrasicki** [reply]
Hi @jasonhenriquez, [I've tried helping, but it didn't work](https://github.com/FreeTubeApp/FreeTube/pull/4379) and it feels like there is something wrong with the way the team makes UI decisions. So I just wanted to ask you to reconsider how those things are done and perhaps consult UI developers/designers from other projects if needed. It's a shame to see such a great project get stuck in this area. The team is doing awesome work, but I think you guys need to change your approach to UI development.

**Reply by kommunarr** [reply]
I think this particular PR was not handled properly by the team, and I apologize for that, but I think it wouldn't be fully honest to over-extrapolate that situation alone. We readily accepted & merged your other PR, and have been working to accept other UX improvement PRs in the time since. This particular situation became unproductive, in my opinion, because both sides got embroiled in individual perspectives & examples on what seemed best to them personally without citing of the actual research & concepts in the UX field. I understand there is the need for improvement, though, and that requires volunteers stepping up to do the work. I would be very receptive to concrete recommendations on what needs to change and/or new discussions & PRs with UX improvements, as broad calls to action can be difficult to translate into material change.

**Reply by pkrasicki** [reply]
My other PR was a simple change for a dropdown and even that required some arguing. So I don't want to make specific suggestions anymore or be involved in this process at all. It's just too difficult. But if you find a designer that you can trust and listen to them, I'm sure that you can solve all of those problems and make this app way better than any competition. You are all great developers, but you need a designer to make design decisions and to make sure the style and UX is consistent on all pages. Otherwise you will keep making the same mistakes.

FreeTube is not the only project with those kinds of issues, but unlike some others, at least it's not limited by its technology. I can't cite UX research and I'm not always good at explaining those concepts. Also, doing that every time wouldn't be productive. All I can do is show you other applications for reference and how they do things. If that's not enough, then there is nothing that I can do other than maintaining my own fork. But that will help only me.

By they way, asking users for their opinion every time is not a solution either, because they are not UI experts. Any UI change means that they have to relearn how to use some feature, so they have a reason to be opposed to any UI change, even if it benefits them in the long term. Fighting this bias is hard and you might have experienced it yourself as a user at some point. When I proposed to redesign the main view, I was asked to implement it as an optional setting, which would be disabled by default. This was done, so that it doesn't upset the users. But then I've noticed that FreeTube has many other settings like this. There are settings to hide the logo or change the background color of the top bar. But users shouldn't have to configure things like that. You should give them good defaults instead.

**Reply by efb4f5ff-1298-471a-8973-3d47447115dc** [reply]
> My other PR was a simple change for a dropdown and even that required some arguing.

Sorry but i dont see any arguing in https://github.com/FreeTubeApp/FreeTube/pull/4405? I only see a suggestions to enhance the PR and a little discussion about the placement of the tooltip.

**Reply by kommunarr** [reply]
I strongly agree with your point about the defaults, and this is the major FLOSS UX trap in a nutshell. See [my discussion here](https://github.com/FreeTubeApp/FreeTube/issues/4594#issuecomment-2054101358). I think we might need to even crack down or at least sequester some of our existing functionality to de-clutter the UI. It's not just about having someone involved who's knowledgeable; it's also ensuring that we have a set of clear design principles that are enforced even when there isn't one set person keeping track of them.

> Sorry but i dont see any arguing [...]

TO @pkrasicki's credit, I think the process felt more drawn out because of how long it took us to review the PR (we were all knee deep in the User Playlist work, which left this PR on the backburner for all of December).

I still would contend that our User Playlist push period was not representative of our processes as a whole (flawed as they are), and I apologize that that caused you to be pushed away. Again, for that specific PR about the side-spacing, that one is very easy for it to fall into subjective territory, where determination of what the "better default" is seemingly arbitrary without more incontrovertible evidence being provided of "how" it is better. We have a lot more glaring issues with FreeTube's UI today, specifically with regards to ease of comprehension and ease of use. We'll need all the support we can get in that process, and to my part, I'll make sure that we can discuss the desired solution more fully in the issue before getting into lots of minor quibbles once the PR is released, which can lead to great frustration for the original PR author. I hope you feel more inclined to contribute to FT to help bring about these improvements that we're reaching towards, as again, we're all just volunteers. Thank you for providing your input & grievances relating to this matter, as it is a very real set of problems that we are actively working on addressing.

**Reply by pkrasicki** [reply]
I appreciate the apology. I didn't know that you were busy with playlists.

>Again, for that specific PR about the side-spacing, that one is very easy for it to fall into subjective territory, where determination of what the "better default" is seemingly arbitrary without more incontrovertible evidence being provided of "how" it is better.

But what kind of evidence could there be? Other than the fact that other modern apps don't do that? The card's background color is an unnecessary decoration (or distraction). Once you remove it, it becomes clear that the padding is not needed. If you have doubts, then my advice is that you should consult another expert.

I don't know how other projects like GNOME (for example) make UI decisions or handle such disagreements, but it might be useful to get in touch with somebody and find out.

<details>
<summary>FreeTube 0.20.0 without card background:</summary>

![freetube-transparent-card](https://github.com/FreeTubeApp/FreeTube/assets/49721209/980049d9-2d44-4a87-b71a-b17e6c2a2958)
</details>

<details>
<summary>Now with other backgrounds removed too, same padding:</summary>

![freetube-transparent-bgs](https://github.com/FreeTubeApp/FreeTube/assets/49721209/7df5a5a8-722e-44b6-a0bf-328544ffe83c)
</details>

**Reply by kommunarr** [reply]
> But what kind of evidence could there be? Other than the fact that other modern apps don't do that? 

People with professional UI/UX design knowledge know how to answer this type of question based off of design principles, UX research, and thorough knowledge of the rationales undergirding the choosing of a given design among the different possibilities available. You can get inspiration for better UX patterns by looking at what products with "good UX" do, but that alone does not provide enough evidence that a design using that UX pattern must be implemented. I can understand if you are not equipped to provide proper justification if you lack the necessary knowledge & experience to do so, but that does not put the impetus on others to take them as a given or validate those claims for you. All that to say, if we did take your PR's suggestion at face value & merge it right in, that modus operandi would be counter to the one you're suggesting in this thread. I apologize if this comes off as negative, but my overall point is that your example that you're using is not serving the overall points you're making, many of which I do agree with. We will put in concentrated effort to improve the user experience for FreeTube going forward, and I hope that work will become very apparent in our upcoming releases.

**Reply by pkrasicki** [reply]
I've never seen anyone work like that, buy maybe it's a good idea to have high standards. Thanks for making UX one of your future priorities.

**Reply by kommunarr** [reply]
Again, thank you for providing your input on this matter. I'll close out this discussion for the time being, but we will keep these concerns in the forefront.
